internal/handlers/pg/pgdb/indexes.go

```diff
@@ -204,8 +204,15 @@ func createPgIndexIfNotExists(ctx context.Context, tx pgx.Tx, schema, table, ind
 			return lazyerrors.Errorf("unknown sort order: %d", field.Order)
 		}
 
-		// It's important to sanitize field.Field data here, as it's a user-provided value.
-		fieldsDef[i] = fmt.Sprintf(`((_jsonb->%s)) %s`, quoteString(field.Field), order)
+		// if the key is foo.bar, then need to modify it to foo -> bar
+		var transformedParts []string
+		fs := strings.Split(field.Field, ".")
+
+		for _, f := range fs {
+			// It's important to sanitize field.Field data here, as it's a user-provided value.
+			transformedParts = append(transformedParts, quoteString(f))
+		}
```

Comment:
```suggestion
		fs := strings.Split(field.Field, ".")
		transformedParts := make([]string, len(fs))

		for i, f := range fs {
			// It's important to sanitize field.Field data here, as it's a user-provided value.
			transformedParts[i] = quoteString(f)
		}
```

Minor suggestion. I think this approach is more consistent with the rest of the codebase. When we know the exact length of the slice, we allocate memory when creating it.